### PR TITLE
feat(tools): csharp codegen uuid support

### PR DESCRIPTION
**Pull request by ragi-dayananda**
# Closes #<CORRESPONDING ISSUE NUMBER>
 

### Changes
UUID support for csharp codegen.  Since UUID's are not supported globally by concerto, added a scalar type UUID with regex validation in `concerto.scalar` namespace.

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 1]
```diff
@@ -228,6 +228,25 @@ class CSharpVisitor {
...
+     * @param {Object} parameters  - the parameter
+     * @return {Object} the result of visiting or null
+     * @private
+     */
+    visitScalarField(field, parameters) {
+        let fieldType;
+        let scalarField = field.getScalarField();
+        if (scalarField.getType() === 'String') {
+            const type = field.getParent().getModelFile().getType(field.getType());
+            if (type.getName() === 'UUID' && type.getNamespace().startsWith('concerto.scalar')) {
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:243]: Better to use `ModelUtil.removeNamespaceVersionFromFullyQualifiedName(field.getFullyQualifiedTypeName()) === 'concerto.scalar.UUID'`

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 2]
```diff
@@ -228,6 +228,25 @@ class CSharpVisitor {
...
+                fieldType = 'UUID';
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:244]: safer to assign `fieldType` to `concerto.scalar.UUID` so we don't lose the namespace in `toCSharpType`.

**Review by dselman — CHANGES_REQUESTED** [review]
Looking good. Couple of tweaks...

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 3]
```diff
@@ -228,6 +228,24 @@ class CSharpVisitor {
...
+     * Visitor design pattern
+     * @param {Field} field - the object being visited
+     * @param {Object} parameters  - the parameter
+     * @return {Object} the result of visiting or null
+     * @private
+     */
+    visitScalarField(field, parameters) {
+        let fieldType;
+        let scalarField = field.getScalarField();
+        if (scalarField.getType() === 'String') {
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:241]: I don't think this check is really necessary. Once we know that the scalar is of type concerto.scalar.UUID we should be good to go...

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js [hunk 4]
```diff
@@ -228,6 +228,24 @@ class CSharpVisitor {
...
+            if(ModelUtil.removeNamespaceVersionFromFullyQualifiedName(field.getFullyQualifiedTypeName()) === 'concerto.scalar.UUID') {
+                fieldType = 'concerto.scalar.UUID';
+            }
+        }
+        return this.writeField(field.getScalarField(), parameters, fieldType);
```
Comment (dselman) [inline packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js:246]: I think you can make this a "one-liner", something like...

```
const fieldType = ModelUtil.removeNamespaceVersionFromFullyQualifiedName(field.getFullyQualifiedTypeName());

return this.writeField(field.getScalarField(), parameters, fieldType === 'concerto.scalar.UUID' ? fieldType : null );
```

**Review by dselman — CHANGES_REQUESTED** [review]
Nit-picks!

File: packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js (hunk 4 above)
Comment (ragi-dayananda) [reply packages/concerto-tools/lib/codegen/fromcto/csharp/csharpvisitor.js]: thanks for the suggestion.

**Review by mttrbrts — APPROVED** [review]

**Review by dselman — APPROVED** [review]
Good work!

**Pull request merged**